### PR TITLE
feat: sdk pay now button enable prop added

### PR DESCRIPTION
**Pull request by sakksham7**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added a prop for SDK Pay now button for enabling and disabling it, until all the validations are checked and correct.
By passing boolean value to that prop the SDK Pay now button can be enabled or disabled.

Before:
Screenshot 2024-08-05 at 6 39 24 PM

After(passing button enable as true):
Screenshot 2024-08-05 at 6 39 45 PM

## How did you test it?

passed value of prop in CheckoutForm.js and then rendered the SDK.

## Checklist

- [X] I ran `npm run re:build`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/Hooks/UtilityHooks.res [hunk 1]
```diff
@@ -11,12 +11,17 @@ let useIsGuestCustomer = () => {
 }
 
 let useHandlePostMessages = (~complete, ~empty, ~paymentType, ~savedMethod=false) => {
+  open RecoilAtoms
+
   let loggerState = Recoil.useRecoilValueFromAtom(RecoilAtoms.loggerAtom)
   let setIsPayNowButtonDisable = RecoilAtoms.payNowButtonDisable->Recoil.useSetRecoilState
```
Comment (PritishBudhiraja) [inline src/Hooks/UtilityHooks.res:17]: If recoilatoms is open remove from these 2 also

File: src/orca-loader/Hyper.res [hunk 2]
```diff
@@ -331,10 +331,14 @@ let make = (publishableKey, options: option<JSON.t>, analyticsInfo: option<JSON.
...
 
-                if (
-                  isSdkButton ||
-                  (val->JSON.Decode.bool->Option.getOr(false) && redirect === "always")
-                ) {
+                if isSdkButton {
+                  if !(val->JSON.Decode.bool->Option.getOr(false)) {
+                    resolve1(json)
+                  } else {
+                    resolve1(data)
```
Comment (PritishBudhiraja) [inline src/orca-loader/Hyper.res:338]: remove resolve in 2 case as I guess it's not needed. @ArushKapoorJuspay

File: src/orca-loader/Hyper.res [hunk 3]
```diff
@@ -331,10 +331,14 @@ let make = (publishableKey, options: option<JSON.t>, analyticsInfo: option<JSON.
...
+                    Window.replace(returnUrl)
+                  }
+                } else if val->JSON.Decode.bool->Option.getOr(false) && redirect === "always" {
                   Window.replace(returnUrl)
                 } else if !(val->JSON.Decode.bool->Option.getOr(false)) {
```
Comment (seekshiva) [inline src/orca-loader/Hyper.res:343]:
```
val->JSON.Decode.bool->Option.getOr(false)
```
this is repeated couple of times here. Can you define it in a variable on top (with a clear name) and use it in the 3 places?

**Review by seekshiva — CHANGES_REQUESTED** [review]

File: src/orca-loader/Hyper.res (hunk 2 above)
Comment (sakksham7) [reply src/orca-loader/Hyper.res:323]: removed resolve1(data)

File: src/orca-loader/Hyper.res (hunk 3 above)
Comment (sakksham7) [reply src/orca-loader/Hyper.res]: got it, did the change

**Review by PritishBudhiraja — DISMISSED** [review]

**Review by PritishBudhiraja — DISMISSED** [review]

**Review by PritishBudhiraja — DISMISSED** [review]

**Review by seekshiva — APPROVED** [review]

**Pull request merged**